src/compressed_tensors/quantization/lifecycle/apply.py

```diff
@@ -104,8 +132,92 @@ def load_pretrained_quantization(model: Module, model_name_or_path: str):
             )
 
 
+def process_transforms_config(
+    transforms_config: TransformationConfig,
+    model: torch.nn.Module,
+    quantization_status: Optional[QuantizationStatus] = QuantizationStatus.INITIALIZED,
+):
+    for _, group in transforms_config.transform_groups.items():
+        # Each group/scheme targets one type of transform
+        transform_type = group.transform_type
+        transform_creation_args = group.transform_creation_args
+
+        # Need a better name - too many groups
+        for transform_arg in group.groups:
+            module_targets = transform_arg.module_targets
+
+            for name, submodule in model.named_modules():
+                if len(transform_arg.ignore) > 0:
+                    if matches := find_name_or_class_matches(
+                        name, submodule, transform_arg.ignore
+                    ):
+                        for match in matches:
+                            print("ignoring", match, name)
+                        continue  # layer matches ignore list, continue
+
+                targets = find_name_or_class_matches(
+                    name, submodule, transform_arg.targets
+                )
+
+                if targets:
+                    # Every layer which matches gets its own transform
+                    # Same transform type and args are used however
+
+                    # attach the transform to the submodule
+                    # because we can have more than one transform, need to attach some
+                    # form of key to fetch
+                    # OR we store it in the dictionary, handle cpu-offloading separatly
+
+                    if hasattr(submodule, "transform_data"):
+                        idx = submodule.transform_data.idx + 1
+                    else:
+                        idx = 0
+                    # only support weight parameters for now, assume one value in
+                    # module targets
+                    transform_name = f"{module_targets[0]}_transform_{idx}"
+
+                    # create an empty tensor OR create a new transform
+                    dtype = getattr(submodule, module_targets[0]).dtype
+                    if quantization_status in [
+                        QuantizationStatus.COMPRESSED,
+                        QuantizationStatus.FROZEN,
+                    ]:
+                        transform = Transforms.load_from_registry(
+                            transform_type,
+                            dtype=dtype,
+                            empty=True,
```

Comment:
Discussing this offline

Reply:
For anyone who's interested: 

As per offline discussion, during decompression (Quantization Status = FROZEN), empty=True is passed in such that empty params pertaining to transforms are created; these params are subsequently filled in (loaded from disk) by the compressor. Resolving this comment!